File: common/src/main/java/com/nivasafinance/common/base/BaseNavigatorService.java
package com.nivasafinance.common.base;

import org.modelmapper.ModelMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.MessageSource;

public class BaseNavigatorService {

    @Autowired
    protected ModelMapper modelMapper;

    @Autowired
    protected MessageSource messageSource;

    public ModelMapper getModelMapper() {
        return modelMapper;
    }

    public void setModelMapper(ModelMapper modelMapper) {
        this.modelMapper = modelMapper;
    }

    public MessageSource getMessageSource() {
        return messageSource;
    }

    public void setMessageSource(MessageSource messageSource) {
        this.messageSource = messageSource;
    }
}

File: common/src/main/java/com/nivasafinance/common/base/BaseNavigatorService.kt
package com.nivasafinance.common.base

import org.modelmapper.ModelMapper
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.MessageSource

open class BaseNavigatorService {

    @Autowired
    lateinit var modelMapper: ModelMapper

    @Autowired
    lateinit var messageSource: MessageSource
}
